src/middleware/bcrypt.ts

```diff
@@ -0,0 +1,15 @@
+import bcrypt from "bcrypt";
+import { IUserBase } from "../interfaces/user";
+import { CREDENTIALS_ERROR } from "../constants/api";
+
+const BcryptMiddelware = {
+    createHash(password: string) {
+        return bcrypt.hash(password, Number(process.env.SALT_ROUNDSs));
+    },
+    async checkPassword(user: IUserBase, password: string) {
+        const result = await bcrypt.compare(password, user.password!);
+        if (result) { return result; } else { throw CREDENTIALS_ERROR; }
```

Comment:
Write   `if (result) { return result; } else { throw CREDENTIALS_ERROR; }` in one line, it is bed. Write as block of code